Sources/Locker/Classes/Locker.swift

```diff
@@ -0,0 +1,210 @@
+                    flags = .touchIDCurrentSet
+                }
+
+                let sacObject = SecAccessControlCreateWithFlags(
+                    kCFAllocatorDefault,
+                    kSecAttrAccessibleWhenUnlockedThisDeviceOnly,
+                    flags,
+                    errorRef
+                )
+
+
+                guard let sacObject = sacObject, errorRef == nil else {
+                    print("can't create sacObject: ", errorRef?.pointee.debugDescription ?? "")
+                    return
+                }
+                let attributes: [CFString : Any] = [
+                    kSecClass : kSecClassGenericPassword,
+                    kSecAttrService : LockerHelpers.keyKeychainServiceName,
+                    kSecAttrAccount : LockerHelpers.keyKeychainAccountName(for: uniqueIdentifier),
+                    kSecValueData : secret.data(using: .utf8) ?? Data(),
+                    kSecUseAuthenticationUI: false,
+                    kSecAttrAccessControl : sacObject
+                ]
+
+                DispatchQueue.global(qos: .default).async {
+                    SecItemAdd(attributes as CFDictionary, nil)
+
+                    // Store current LA policy domain state
+                    LockerHelpers.storeCurrentLAPolicyDomainState()
+                }
+            }
+        }
+    }
+
+    public static func retrieveCurrentSecret(for uniqueIdentifier: String, operationPrompt: String, success: ((String?) -> Void)?, failure: ((OSStatus) -> Void)?) {
+        if TARGET_OS_SIMULATOR != 0 {
+            let simulatorSecret = Locker.userDefaults?.string(forKey: uniqueIdentifier)
+            if simulatorSecret == nil {
+                failure?(errSecItemNotFound)
+                return
+            }
+            success?(simulatorSecret)
+        } else {
+            let query: [CFString : Any] = [
+                kSecClass : kSecClassGenericPassword,
+                kSecAttrService : LockerHelpers.keyKeychainServiceName,
+                kSecAttrAccount : LockerHelpers.keyKeychainAccountName(for: uniqueIdentifier),
+                kSecMatchLimit : kSecMatchLimitOne,
+                kSecReturnData : true,
+                kSecUseOperationPrompt : operationPrompt
+            ]
+
+            DispatchQueue.global(qos: .default).async {
+                var dataTypeRef: CFTypeRef? = nil
+
+                let status = SecItemCopyMatching(query as CFDictionary, &dataTypeRef)
+                if status == errSecSuccess {
+                    if let success = success {
+                        guard let resultData = dataTypeRef as? Data else {
+                            failure?(errSecItemNotFound)
+                            return
+                        }
+                        let result = String(data: resultData, encoding: .utf8)
+                        DispatchQueue.main.async {
+                            success(result)
+                        }
+                    }
```

Comment:
Added it to the guard statement above, changed the closure to take String, and a secret above was added to the guard statement. That's a much better approach since we all use success when we get something from userDefaults and we can only set a non-nil value as a secret 😄